File: backend/app/schemas/stocks.py
from decimal import Decimal

from pydantic import BaseModel


class Quote(BaseModel):
    ticker: str
    price: Decimal
    change: Decimal
    change_pct: Decimal
    volume: int
    latest_trading_day: str


class PriceBar(BaseModel):
    date: str  # ISO date string — chart display only, floats acceptable here
    open: float
    high: float
    low: float
    close: float
    volume: int


class CompanyOverview(BaseModel):
    ticker: str
    name: str
    market_cap: int | None = None
    week_52_high: Decimal | None = None
    week_52_low: Decimal | None = None
    description: str | None = None
    sector: str | None = None
    industry: str | None = None


class SearchResult(BaseModel):
    symbol: str
    name: str
    type: str
    region: str

File: frontend/src/types/stocks.ts
/** Decimal fields from the backend serialise as strings in JSON. */
export interface Quote {
  ticker: string;
  price: string;
  change: string;
  change_pct: string;
  volume: number;
  latest_trading_day: string;
}

export interface PriceBar {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
}

export interface CompanyOverview {
  ticker: string;
  name: string;
  market_cap: number | null;
  week_52_high: string | null;
  week_52_low: string | null;
  description: string | null;
  sector: string | null;
  industry: string | null;
}

export interface SearchResult {
  symbol: string;
  name: string;
  type: string;
  region: string;
}
